showkase/src/main/java/com/airbnb/android/showkase/ui/ShowkaseBrowserApp.kt

```diff
@@ -308,6 +314,24 @@ internal fun ShowkaseBodyContent(
     }
 }
 
+private fun getCategoryMetadataMap(
+    groupedComponentMap: Map<String, List<ShowkaseBrowserComponent>>,
+    groupedColorsMap: Map<String, List<ShowkaseBrowserColor>>,
+    groupedTypographyMap: Map<String, List<ShowkaseBrowserTypography>>,
+): Map<ShowkaseCategory, Int> {
+    val componentCount = groupedComponentMap.values.flatCount()
+    val colorsCount = groupedColorsMap.values.flatCount()
+    val typographyCount = groupedTypographyMap.values.flatCount()
+    
+    return mapOf(
+        ShowkaseCategory.COMPONENTS to componentCount,
+        ShowkaseCategory.COLORS to colorsCount,
+        ShowkaseCategory.TYPOGRAPHY to typographyCount
+    )
+}
+
+internal fun Collection<List<*>>.flatCount() = flatMap { it }.count()
```

Comment:
since this is so specialized it is unlikely to get used elsewhere anyway. Might as well make the receiver `Map<String, List<*>>` so you don't have to repeat the `values` part, and then make it a local function